### PR TITLE
Importing chunks from the CR Processing Spreadsheet

### PR DESCRIPTION
**Pull request by ahmad-elbatanouni**
**Checking whether a cell is modified** hasn't been implemented yet .. I need help with the keywords to look for regarding this point.

-- Also please note **typescript** version hasn't been updated in the `package.json` file... I didn't update it as you mentioned you'll handle this by yourself.

### CODE REVIEW

**Comment by ahmad-elbatanouni** [comment]
[Importing chunks from the CR Processing Spreadsheet](https://trello.com/c/VG4HHpZA/54-importing-chunks-from-the-cr-processing-spreadsheet)

File: README.md [hunk 1]
```diff
@@ -53,6 +53,7 @@ $ firebase functions:config:set coordinator.email_address='EMAIL'
 $ firebase functions:config:set coordinator.timeZoneOffset=NUMBER of HOURs
 #Importing a spreadsheet to the database variables
 $ firebase functions:config:set sqr.spreadsheetId='Google Spreadsheet ID'
+firebase functions:config:set sound_editing.spreadsheet_id="Google Spreadsheet ID"
```
Comment (anantakrishna) [inline README.md]: There is no “Sound Editing Spreadsheet”. This one should be called “CR Processing Spreadsheet”.

```suggestion
 firebase functions:config:set reporting.content.processing.spreadsheet_id="Google Spreadsheet ID"
```

File: functions/src/modules/SE.functions.ts [hunk 2]
```diff
@@ -88,3 +89,227 @@ export const processNewAllotment = functions.database
     });
     return 1;
   });
+
+
+const timeToMins = (time) => {
```
Comment (anantakrishna) [inline functions/src/modules/SE.functions.ts]: Add documentation to this function, would help.

File: functions/src/modules/SE.functions.ts [hunk 3]
```diff
@@ -88,3 +89,227 @@ export const processNewAllotment = functions.database
...
+    let hours = time.split(':')[0];
+    let mins = time.split(':')[1];
+    return (+hours * 60) + (+mins);
+}
+
+export const importSpreadSheetData = functions.runWith({
```
Comment (anantakrishna) [inline functions/src/modules/SE.functions.ts]:
```suggestion
 export const importChunks = functions.runWith({
```

File: functions/src/modules/SE.functions.ts [hunk 4]
```diff
@@ -88,3 +89,227 @@ export const processNewAllotment = functions.database
...
+          noResolution: 0, 
+          overlappingChunks: [],
+          modified: []
+      };
+
+  for (let i = 0; i < sheetTitles.length; i++) {
+      const result = await sheets
+          .spreadsheets.values.get({
+              spreadsheetId,
+              range: `${sheetTitles[i]}!A2:AZ1000`
```
Comment (anantakrishna) [inline functions/src/modules/SE.functions.ts]: Isn't there another way to get all the sheet? Some lists will have more than 1000 rows… So think of another, more reliable way.

File: functions/src/modules/SE.functions.ts [hunk 5]
```diff
@@ -88,3 +89,227 @@ export const processNewAllotment = functions.database
...
+          chunks: [], 
+          skip: false, 
+          track: null 
+      };
+
+      for (const row of rows) {
+          let skip = false;
+
+          // Misplaced audiofile entry
+          if (row[columnsIndex['AudioFileName']].split('-')[0] != sheetTitles[i]) {
```
Comment (anantakrishna) [inline functions/src/modules/SE.functions.ts]: We had a regex for extracting list from the file name, no?

File: functions/src/modules/SE.functions.ts [hunk 6]
```diff
@@ -88,3 +89,227 @@ export const processNewAllotment = functions.database
...
+              lastFile.skip = skip;
+          }
+      }
+  }
+
+  //////////////////////////////////////
+  // Notifying the admin of the results
+  //////////////////////////////////////
+  await db.ref(`/email/notifications`).push({
+      template: 'importing-se-chunks',
```
Comment (anantakrishna) [inline functions/src/modules/SE.functions.ts]:
```suggestion
       template: 'chunks-import-summary',
```

File: functions/src/modules/SE.functions.ts [hunk 7]
```diff
@@ -88,3 +89,227 @@ export const processNewAllotment = functions.database
...
+                  beginning: row[columnsIndex['Beginning']],
+                  ending: row[columnsIndex['Ending']],
+              }];
+              lastFile.skip = skip;
+          }
+      }
+  }
+
+  //////////////////////////////////////
+  // Notifying the admin of the results
```
Comment (anantakrishna) [inline functions/src/modules/SE.functions.ts]:
```suggestion
   // Notifying the coordinator of the results
```

File: functions/src/modules/SE.functions.ts [hunk 8]
```diff
@@ -88,3 +89,227 @@ export const processNewAllotment = functions.database
...
+  //////////////////////////////////////
+  await db.ref(`/email/notifications`).push({
+      template: 'importing-se-chunks',
+      to: functions.config().coordinator.email_address,
+      bcc: [{ email: functions.config().coordinator.email_address }],
```
Comment (anantakrishna) [inline functions/src/modules/SE.functions.ts]: No need, as the coordinator is already in To field.

File: functions/src/modules/SE.functions.ts (hunk 4 above)
Comment (ahmad-elbatanouni) [reply functions/src/modules/SE.functions.ts]: The maximum number of rows that can be obtained in one call is 1000 ... Is it okay if I get the data in multiple calls ?

Comment (ahmad-elbatanouni) [reply functions/src/modules/SE.functions.ts]: @anantakrishna

**Comment by anantakrishna** [comment]
Yes.

**Pull request closed without merging**